Realm.Shared/Dynamic/DynamicRealmObjectHelper.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using Realms.Weaving;
+
+namespace Realms.Dynamic
+{
+    internal class DynamicRealmObjectHelper : IRealmObjectHelper
```

Comment:
I have no problems with fields. Is there any advantage to a property here besides from binary signature compatibility in case we would ever need to switch to a property (in which case, all uses will be recompiled as well, so not really an issue)?